tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs

```diff
@@ -75,6 +75,14 @@ private string RunQuery(UserQuery query)
             queryDict.Add(LabApiConstants.MobileDeviceManagementWithConditionalAccess, LabApiConstants.False);
 
             //Building user query
+            if (!string.IsNullOrWhiteSpace(query.Upn))
+            {
+                queryDict.Add(LabApiConstants.Upn, query.Upn);
+
+                uriBuilder.Query = string.Join("&", queryDict.Select(x => x.Key + "=" + x.Value.ToString()));
+                return _httpClient.GetStringAsync(uriBuilder.ToString()).GetAwaiter().GetResult();
```

Comment:
why escape here and not at the escape point below? If this is needed consider having a method named RunUserQuery, and refactor code for re-use between these two mehtods.

Reply:
when you specify the upn when sending a query to the lab, there is no reason to continue adding other query parameters because the lab endpoint does not need to search for anything. it will simply return to you the upn you are requesting.